uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java

```diff
@@ -0,0 +1,196 @@
+        BROKER_API("broker_api", "Broker API"),
+        FLIGHTS_API("broker_flights", "Broker Flights API");
+
+        @Getter
+        private final String resourceId;
+        @Getter
+        private final String title;
+    }
+
+    /**
+     * This method clicks on the button with the given resource id
+     *
+     * @param resourceIdButton the resource id of the button to be clicked
+     */
+    public void clickButton(@NonNull final String resourceIdButton) {
+        final String resourceId = CommonUtils.getResourceId(
+                BROKER_HOST_APP_PACKAGE_NAME,
+                resourceIdButton
+        );
+        UiAutomatorUtils.handleButtonClick(resourceId);
+    }
+
+    /**
+     * This method fills the text box with the given resource id with the given text
+     *
+     * @param resourceEditText the resource id of the text box
+     * @param text             the text to be write in the text box
+     */
+    public void fillTextBox(@NonNull final String resourceEditText, @NonNull final String text) {
+        final String resourceId = CommonUtils.getResourceId(
+                BROKER_HOST_APP_PACKAGE_NAME,
+                resourceEditText
+        );
+        UiAutomatorUtils.handleInput(resourceId, text);
+    }
+
+    /**
+     * This method reads the text from the text box with the given resource id
+     *
+     * @param resourceEditText the resource id of the text box
+     * @return the text from the text box
+     */
+    public String readTextBox(@NonNull final String resourceEditText) {
+        final String resourceId = CommonUtils.getResourceId(
+                BROKER_HOST_APP_PACKAGE_NAME,
+                resourceEditText
+        );
+        final UiObject textBox = UiAutomatorUtils.obtainUiObjectWithResourceId(resourceId);
+        try {
+            return textBox.getText();
+        } catch (final UiObjectNotFoundException e) {
+            throw new AssertionError("Could not read text from text box", e);
+        }
+    }
+
+    /**
+     * This method dismisses the dialog box and returns the text from the dialog box
+     *
+     * @return the text from the dialog box
+     */
+    static public String dismissDialogBoxAndGetText() {
+        // Look for the dialog box
+        final UiObject dialogBox = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                DIALOG_BOX_RESOURCE_ID
+        );
+        Assert.assertTrue("Assert dialog box", dialogBox.exists());
+        try {
+            final String dialogBoxText = dialogBox.getText();
+            final String[] dialogBoxSplit = dialogBoxText.split(":");
+            return dialogBoxSplit.length > 1 ? dialogBoxSplit[1] : dialogBoxText;
+        } catch (final UiObjectNotFoundException e) {
+            throw new AssertionError(e);
+        } finally {
+            // dismiss dialog
+            UiAutomatorUtils.handleButtonClick(DIALOG_BOX_OK_BUTTON_RESOURCE_ID);
+        }
+    }
+
+    static public void delay(final long delayInSeconds) {
```

Comment:
There is a method in ThreadUtils.java 'sleepSafely', we can use that.

Reply:
done